src/nodejs/hooks/seeker_agent_test.go

```diff
@@ -0,0 +1,479 @@
+				contents, err := ioutil.ReadFile(entryPointPath)
+				Expect(err).To(BeNil())
+				Expect(string(contents)).To(Equal(
+					"require('./seeker/node_modules/@synopsys-sig/seeker-inline');\n" +
+						"some mock javascript code\n"))
+			})
+			It("Fails to prepend the require to the server.js file - when the file does not exist", func() {
+				entryPointPath := filepath.Join(buildDir, "server.js")
+				Expect(entryPointPath).ToNot(BeAnExistingFile())
+				err = seeker.AfterCompile(stager)
+				Expect(err).NotTo(BeNil())
+				Expect(err.Error()).To(ContainSubstring("no such file or directory"))
+			})
+
+		})
+	})
+})
+
+func assertSensorDownload() {
+	Expect(httpmock.GetCallCountInfo()["GET "+getVersionURL()]).To(Equal(1))
+	Expect(httpmock.GetCallCountInfo()["GET "+getSensorURL()]).To(Equal(1))
+	Expect(httpmock.GetCallCountInfo()).ShouldNot(ContainElement("GET " + getAgentURL()))
+	Expect(httpmock.GetTotalCallCount()).To(Equal(2))
+}
+func assertAgentDownload() {
+	Expect(httpmock.GetCallCountInfo()["GET "+getVersionURL()]).To(Equal(1))
+	Expect(httpmock.GetCallCountInfo()["GET "+getAgentURL()]).To(Equal(1))
+	Expect(httpmock.GetCallCountInfo()).ShouldNot(ContainElement("GET " + getSensorURL()))
+	Expect(httpmock.GetTotalCallCount()).To(Equal(2))
+}
+
+func getAgentZip() ([]byte, error) {
+	return getFixtureContent("NODEJS_agent.zip")
+}
+func getSensorZip() ([]byte, error) {
+	return getFixtureContent("NODEJS_SensorWithAgent.zip")
+}
+func getFixtureContent(fileName string) ([]byte, error) {
+	path, err := filepath.Abs("../../../fixtures/seeker/" + fileName)
+	if err != nil {
+		return nil, err
+	}
+	fileContent, err := ioutil.ReadFile(path)
+	return fileContent, err
+}
+
+func mockSeekerVersionThatSupportsOnlySensorDownload() {
+	httpmock.RegisterResponder("GET", getVersionURL(),
+		httpmock.NewStringResponder(200, `{"publicName":"Seeker Enterprise Server","version":"2018.05-SP1-SNAPSHOT","buildNumber":"20180629131550","scmBranch":"origin/release/v2018.06","scmRevision":"815ba309"}`))
+}
+
+func mockSeekerVersionThatSupportsAgentDownload() {
+	httpmock.RegisterResponder("GET", getVersionURL(),
+		httpmock.NewStringResponder(200, `{"publicName":"Seeker Enterprise Server","version":"2018.06-SNAPSHOT","buildNumber":"20180629131550","scmBranch":"origin/release/v2018.06","scmRevision":"815ba309"}`))
+}
+func mockSensorDownload() {
+	zipContent, _ := getSensorZip()
+	httpmock.RegisterResponder("GET", getSensorURL(),
+		httpmock.NewBytesResponder(200, zipContent))
+}
+
+func mockAgentDownload() {
+	zipContent, _ := getAgentZip()
+	httpmock.RegisterResponder("GET", getAgentURL(),
+		httpmock.NewBytesResponder(200, zipContent))
+}
+func getSensorURL() string {
+	return "http://10.120.9.117:8082/rest/ui/installers/binaries/LINUX"
+}
+func getAgentURL() string {
+	return "http://10.120.9.117:8082/rest/ui/installers/agents/binaries/NODEJS"
+}
+func getVersionURL() string {
+	return "http://10.120.9.117:8082/rest/api/version"
```

Comment:
It would be helpful if you would factor out these static IPs into consts.

Reply:
fixed